src/lib.rs

```diff
@@ -43,6 +43,35 @@ fn __system_time_size_check() {
 }
 
 
+/// Call this function on startup to ensure that all assumptions about the platform are valid.
```

Comment:
nit: Can you document a bit better what they are, and that they are not believed to be violated on any supported rust platforms, and, thus, users should really, really just call this cause its cheap and fast and ensures they're sane.